openedx/core/djangoapps/user_authn/views/logout.py

```diff
@@ -75,10 +77,13 @@ def target(self):
     def dispatch(self, request, *args, **kwargs):
         # We do not log here, because we have a handler registered to perform logging on successful logouts.
         request.is_from_logout = True
-
+        if third_party_auth.is_enabled() and pipeline.running(request):
+            running_pipeline = pipeline.get(request)
+            self.third_party_provider = provider.Registry.get_from_pipeline(running_pipeline)
+            self.auth_backend = request.session._session_cache.get('social_auth_last_login_backend', '').encode()
```

Comment:
why we need to set this `self.auth_backend`?

Reply:
To compare with SAML_BACKEND `tpa-saml`. 

**L156** `if referrer == LEARNING_PORTAL and self.auth_backend == SAML_BACKEND:`